packages/backend/src/services/warehouseClients/PostgresWarehouseClient.ts

```diff
@@ -1,8 +1,95 @@
-import { CreatePostgresCredentials, CreateRedshiftCredentials } from 'common';
+import {
+    CreatePostgresCredentials,
+    CreateRedshiftCredentials,
+    DimensionType,
+} from 'common';
 import * as pg from 'pg';
 import { WarehouseConnectionError, WarehouseQueryError } from '../../errors';
 import { QueryRunner } from '../../types';
 
+export enum PostgresTypes {
+    INTEGER = 'integer',
+    INT = 'int',
+    INT2 = 'int2',
+    INT4 = 'int4',
+    INT8 = 'int8',
+    MONEY = 'money',
+    SMALLSERIAL = 'smallserial',
+    SERIAL = 'serial',
+    SERIAL2 = 'serial2',
+    SERIAL4 = 'serial4',
+    SERIAL8 = 'serial8',
+    BIGSERIAL = 'bigserial',
+    BIGINT = 'bigint',
+    SMALLINT = 'smallint',
+    BOOLEAN = 'boolean',
+    BOOL = 'bool',
+    DATE = 'date',
+    DOUBLE_PRECISION = 'double precision',
+    FLOAT = 'float',
+    FLOAT4 = 'float4',
+    FLOAT8 = 'float8',
+    JSON = 'json',
+    JSONB = 'jsonb',
+    NUMERIC = 'numeric',
+    DECIMAL = 'decimal',
+    REAL = 'real',
+    CHAR = 'char',
+    CHARACTER = 'character',
+    NCHAR = 'nchar',
+    BPCHAR = 'bpchar',
+    VARCHAR = 'varchar',
+    CHARACTER_VARYING = 'character varying',
+    NVARCHAR = 'nvarchar',
+    TEXT = 'text',
+    TIME = 'time',
+    TIME_TZ = 'timetz',
+    TIME_WITHOUT_TIME_ZONE = 'time without time zone',
+    TIMESTAMP = 'timestamp',
+    TIMESTAMP_TZ = 'timestamptz',
+    TIMESTAMP_WITHOUT_TIME_ZONE = 'timestamp without time zone',
+}
+
+const mapFieldType = (type: string): DimensionType => {
+    switch (type) {
+        case PostgresTypes.DECIMAL:
+        case PostgresTypes.NUMERIC:
+        case PostgresTypes.INTEGER:
+        case PostgresTypes.MONEY:
+        case PostgresTypes.SMALLSERIAL:
+        case PostgresTypes.SERIAL:
+        case PostgresTypes.SERIAL2:
+        case PostgresTypes.SERIAL4:
+        case PostgresTypes.SERIAL8:
+        case PostgresTypes.BIGSERIAL:
+        case PostgresTypes.INT2:
+        case PostgresTypes.INT4:
+        case PostgresTypes.INT8:
+        case PostgresTypes.BIGINT:
+        case PostgresTypes.SMALLINT:
+        case PostgresTypes.FLOAT:
+        case PostgresTypes.FLOAT4:
+        case PostgresTypes.FLOAT8:
+        case PostgresTypes.DOUBLE_PRECISION:
+        case PostgresTypes.REAL:
+            return DimensionType.NUMBER;
+        case PostgresTypes.DATE:
+            return DimensionType.DATE;
+        case PostgresTypes.TIME:
+        case PostgresTypes.TIME_TZ:
+        case PostgresTypes.TIMESTAMP:
+        case PostgresTypes.TIMESTAMP_TZ:
```

Comment:
In the initial design of this, I rejected `TIMESTAMP_TZ` types with the timezone embedded since we don't handle timezones at all. For simplicity, we could accept only `TIMESTAMP` with no time zone, assume it's UTC, and then the frontend renders the time according to the `locale` of the browser.

If we accept timestamp_tz, do we need to do some conversion?

Reply:
Haven't thought about the timezones but I also don't understand what we lose mapping it to the timestamp type. 
If we don't match that column type to timestamp it will default to string which provides even worse filtering.

Reply:
Well if the backend only works in UTC, then the frontend can render all timestamps in the browsers locale setting. The tricky bit is that when timestamps arrive at the server with a time offset, we need to convert it to UTC. I previously got around this by simply not accepting timestamps with offsets.

Perhaps this is a separate ticket. If we want to handle TIMESTAMP_TZ we should have some logic to convert to UTC.

Reply:
> Perhaps this is a separate ticket. If we want to handle TIMESTAMP_TZ we should have some logic to convert to UTC.

Agree.
Should we keep this mapping then ? is not the perfect solution but is one step closer since right now we map it to a string.

Reply:
Yep! https://github.com/lightdash/lightdash/issues/718